storybook/main.js

```diff
@@ -30,20 +28,11 @@ module.exports = {
 		emotionAlias: false,
 		storyStoreV7: true,
 	},
-	// Workaround:
-	// https://github.com/storybookjs/storybook/issues/12270
-	webpackFinal: async ( config ) => {
-		// Find the DefinePlugin.
-		const plugin = config.plugins.find( ( p ) => {
-			return p.definitions && p.definitions[ 'process.env' ];
-		} );
-		// Add custom env variables.
-		Object.keys( customEnvVariables ).forEach( ( key ) => {
-			plugin.definitions[ 'process.env' ][ key ] = JSON.stringify(
-				customEnvVariables[ key ]
-			);
-		} );
-
-		return config;
-	},
+	env: ( config ) => ( {
+		...config,
+		// Pretend we're in the Gutenberg plugin.
+		// Disable reason: We're defining IS_GUTENBERG_PLUGIN, not using it.
+		// eslint-disable-next-line @wordpress/is-gutenberg-plugin
+		IS_GUTENBERG_PLUGIN: true,
```

Comment:
I can still see the error about registering a module more than once in Storybook:

![Screenshot 2023-02-03 at 16 59 24](https://user-images.githubusercontent.com/1083581/216649162-2b911571-75ad-4bdb-82e5-ad38911bcd51.png)


Should we set the `ALLOW_EXPERIMENT_REREGISTRATION` flag here as well? Is setting the `IS_GUTENBERG_PLUGIN` variable still necessary?

Reply:
Strangely, I wasn't seeing that error locally. But I agree with you! Fixed in [22bd457](https://github.com/WordPress/gutenberg/pull/47229/commits/22bd4575b7e99350cf4ffdac97e5bd58a774282e).